lib/msf/core/post/windows.rb

```diff
@@ -1,4 +1,18 @@
 # -*- coding: binary -*-
 
 module Msf::Post::Windows
+  # Escape a string literal value to be included as an argument to cmd.exe. The escaped value *should not* be placed
+  # within double quotes as this will alter now it is evaluated (e.g. `echo "^"((^&test) Foo^""` is different than
+  # `echo ^"((^&test) Foo^"`.
+  #
+  # @param [String] string The string to escape for use with cmd.exe.
+  # @param [Boolean] spaces Whether or not to escape spaces. If the string is being passed to echo, set this to false
+  #   otherwise if it's an argument, set it to true.
+  # @return [String] The escaped string.
+  def self.escape_cmd_literal(string, spaces:)
```

Comment:
`spaces` is optional here because sometimes they need to be escaped, and other times they do not. 

This snippet demonstrates the issue. A small Python stub is used to print the arguments then `one" "argument` is passed to it showing that the space is kept but when passed to echo, the double quotes are kept.
```
Microsoft Windows [Version 10.0.17763.4377](c) 2018 Microsoft Corporation. All rights reserved.
msflab\smcintyre@DC C:\Users\smcintyre>type args.py
import sys; print(repr(sys.argv))
msflab\smcintyre@DC C:\Users\smcintyre>python args.py one" "argument
['args.py', 'one argument']
msflab\smcintyre@DC C:\Users\smcintyre>echo one" "argument 
one" "argument
msflab\smcintyre@DC C:\Users\smcintyre>
```

Allowing the user to specify means the escaped value can be used in both contexts. Psexec should *not* escape spaces because the command is passed to echo.